core/new-gui/src/app/workspace/component/workflow-editor/workflow-editor.component.spec.ts

```diff
@@ -859,5 +872,69 @@ describe("WorkflowEditorComponent", () => {
       expect(jointGraphWrapper.getCurrentHighlightedOperatorIDs()).toContain(mockScanPredicate.operatorID);
       expect(jointGraphWrapper.getCurrentHighlightedOperatorIDs()).toContain(mockResultPredicate.operatorID);
     });
+
+    //undo
+    it("should undo action when press command + Z or control + Z", () => {
+      spyOn(workflowVersionService, "getDisplayParticularVersionStream").and.returnValue(of(false));
+      spyOn(workflowCollabService, "isLockGranted").and.returnValue(true);
+      spyOn(undoRedoService, "canUndo").and.returnValue(true);
+      let undoSpy = spyOn(undoRedoService, "undoAction");
+      fixture.detectChanges();
+      const event1 = new KeyboardEvent("keydown", { key: "Z", metaKey: true, shiftKey: false });
+      (document.activeElement as HTMLElement)?.blur();
+      document.dispatchEvent(event1);
+      fixture.detectChanges();
+      setTimeout(() => {
```

Comment:
the main concern is the `setTimeout` here, because this will execute **after** the whole function exits, so basically this will be: 1) press meta+Z, 2) press ctrl+Z, 3) check, 4) check, can we remove the `setTimeout`?